projects/plugins/jetpack/extensions/blocks/ai-chat/use-submit-question.js

```diff
@@ -12,22 +12,55 @@ import { referenceTexts, initialWaitTexts } from './response-strings';
 
 const blogId = '9619154';
 
-export default function useSubmitQuestion() {
+export default function useSubmitQuestion( setAiResponse ) {
 	const [ question, setQuestion ] = useState( '' );
-
-	const [ answer, setAnswer ] = useState();
 	const [ references, setReferences ] = useState( [] );
 	const [ isLoading, setIsLoading ] = useState( false );
 	const [ waitString, setWaitString ] = useState( '' );
+	const [ callOpenAI, setCallOpenAI ] = useState( false );
 	const [ error, setError ] = useState();
 
 	const handleAPIError = () => {
 		setReferences( [] );
-		setAnswer( undefined );
+		setAiResponse( undefined );
 		setIsLoading( false );
 		setError( __( 'I seem to have encountered a problem. Please try again.', 'jetpack' ) );
 	};
 
+	useEffect( () => {
+		let source;
+		if ( callOpenAI === true ) {
+			const encoded = encodeURIComponent( question );
+			const path = addQueryArgs(
+				`https://public-api.wordpress.com/wpcom/v2/sites/${ blogId }/jetpack-search/ai/search`,
+				{
+					stop_at: 'response',
+					query: encoded,
+					stream: true,
+				}
+			);
+			source = new EventSource( path );
+
+			source.addEventListener( 'open', () => {
```

Comment:
This streaming of answers from OpenAI is dependent on D109907 being merged and an additional `stream` param being piped through API calls